torchvision/transforms/functional.py

```diff
@@ -388,41 +396,57 @@ def pad(img: Tensor, padding: List[int], fill: int = 0, padding_mode: str = "con
     return F_t.pad(img, padding=padding, fill=fill, padding_mode=padding_mode)
 
 
-def crop(img, top, left, height, width):
-    """Crop the given PIL Image.
+def crop(img: Tensor, top: int, left: int, height: int, width: int) -> Tensor:
+    """Crop the given image at specified location and output size.
+    The image can be a PIL Image or a Tensor, in which case it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading
+    dimensions
 
     Args:
-        img (PIL Image): Image to be cropped. (0,0) denotes the top left corner of the image.
+        img (PIL Image or Tensor): Image to be cropped. (0,0) denotes the top left corner of the image.
         top (int): Vertical component of the top left corner of the crop box.
         left (int): Horizontal component of the top left corner of the crop box.
         height (int): Height of the crop box.
         width (int): Width of the crop box.
 
     Returns:
-        PIL Image: Cropped image.
+        PIL Image or Tensor: Cropped image.
     """
-    if not _is_pil_image(img):
-        raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
 
-    return img.crop((left, top, left + width, top + height))
+    if not isinstance(img, torch.Tensor):
+        return F_pil.crop(img, top, left, height, width)
 
+    return F_t.crop(img, top, left, height, width)
 
-def center_crop(img, output_size):
-    """Crop the given PIL Image and resize it to desired size.
+
+def center_crop(img: Tensor, output_size: List[int]) -> Tensor:
+    """Crops the given image at the center.
+    The image can be a PIL Image or a Tensor, in which case it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading dimensions
 
     Args:
-        img (PIL Image): Image to be cropped. (0,0) denotes the top left corner of the image.
-        output_size (sequence or int): (height, width) of the crop box. If int,
-            it is used for both directions
+        img (PIL Image or Tensor): Image to be cropped.
+        output_size (sequence or int): (height, width) of the crop box. If int or sequence with single int
+            it is used for both directions.
+
     Returns:
-        PIL Image: Cropped image.
+        PIL Image or Tensor: Cropped image.
     """
     if isinstance(output_size, numbers.Number):
         output_size = (int(output_size), int(output_size))
-    image_width, image_height = img.size
+    elif isinstance(output_size, (tuple, list)) and len(output_size) == 1:
+        output_size = (output_size[0], output_size[0])
+
+    image_width, image_height = _get_image_size(img)
     crop_height, crop_width = output_size
-    crop_top = int(round((image_height - crop_height) / 2.))
-    crop_left = int(round((image_width - crop_width) / 2.))
+
+    # crop_top = int(round((image_height - crop_height) / 2.))
+    # Result can be different between python func and scripted func
```

Comment:
Can you explain why the results can be different? If yes this is a bug that should be reported / fixed

Reply:
This is, probably, a bug with `torch.jit.script`:
```python
import torch

def foo(x: int) -> int:    
    return int(round(x / 2.0))

sfoo = torch.jit.script(foo)

assert foo(7) == sfoo(7)
# but
assert foo(5) == sfoo(5)
> 2 vs 3
```

Reply:
Thanks for the snippet, this can be further simplified as follows
```python
import torch

def foo():    
    return round(2.5)

sfoo = torch.jit.script(foo)
print(foo(), sfoo())
# 2, 3.0
# different values and types
```

Reply:
Opened an issue in PyTorch to track this https://github.com/pytorch/pytorch/issues/40771

Can you add a comment to revert back to the original implementation once that issue is fixed?

Reply:
Actually, I saw that you left the code commented out, so this is fine for now